rskj-core/src/test/java/co/rsk/peg/whitelist/WhitelistStorageProviderImplTest.java

```diff
@@ -0,0 +1,176 @@
+    void save_whenLockWhiteListIsNotNull_ShouldReturnSavedValue() {
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        //make sure the lockWhitelist is empty
+        assertEquals(0, lockWhitelist.getAll().size());
+
+        LockWhitelistEntry oneOffWhiteListEntry =  addOneOffWhiteListEntry();
+        lockWhitelist.put(firstBtcAddress, oneOffWhiteListEntry);
+
+        whitelistStorageProvider.save(activationConfig);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        assertEquals(1, actualLockWhitelist.getAll().size());
+
+        //Making sure the saved value is correct
+        Map<Address, OneOffWhiteListEntry> whitelistedAddresses = getAddressFromOneOffWhiteListEntryInMemoryStorage();
+        assertTrue(whitelistedAddresses.containsKey(firstBtcAddress));
+    }
+
+    @Test
+    void save_whenIsActiveRSKIP87_ShouldSavedUnlimitedLockWhitelist() {
+        when(activationConfig.isActive(ConsensusRule.RSKIP87)).thenReturn(true);
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        //make sure the lockWhitelist is empty
+        assertEquals(0, lockWhitelist.getAll().size());
+
+        LockWhitelistEntry oneOffWhiteListEntry =  addOneOffWhiteListEntry();
+        LockWhitelistEntry unlimitedWhiteListEntry = addUnlimitedWhiteListEntry();
+        lockWhitelist.put(firstBtcAddress, oneOffWhiteListEntry);
+        lockWhitelist.put(secondBtcAddress, unlimitedWhiteListEntry);
+
+        whitelistStorageProvider.save(activationConfig);
+
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        assertEquals(2, actualLockWhitelist.getAll().size());
+
+        //Making sure the saved value is correct
+        Map<Address, OneOffWhiteListEntry> oneOffWhiteListEntryMap = getAddressFromOneOffWhiteListEntryInMemoryStorage();
+        assertTrue(oneOffWhiteListEntryMap.containsKey(firstBtcAddress));
+
+        Map<Address, LockWhitelistEntry> lockWhitelistEntryMap = getAddressFromUnlimitedWhiteListEntryInMemoryStorage();
+        assertTrue(lockWhitelistEntryMap.containsKey(secondBtcAddress));
+    }
+
+    @Test
+    void getLockWhitelist_whenLockWhitelistIsNull_shouldReturnZeroEntries() {
+        LockWhitelist actualLockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+
+        assertEquals(0, actualLockWhitelist.getAll().size());
+    }
+
+    @Test
+    void getLockWhitelist_whenLockWhitelistIsNotNull_shouldReturnOneEntry() {
+        LockWhitelist lockWhitelist = whitelistStorageProvider.getLockWhitelist(activationConfig, networkParameters);
+        LockWhitelistEntry oneOffWhiteListEntry =  addOneOffWhiteListEntry();
+        lockWhitelist.put(firstBtcAddress, oneOffWhiteListEntry);
```

Comment:
I don't think this is properly set up. You are getting the LockWhitelist object from the storage provider and modifying the object reference.

See how something similar was implemented here https://github.com/rsksmart/rskj/blob/federation-support-refactor-integration/rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java#L662

Basically store the value directly in the storage and then assert getting it

Reply:
Done!